src/test/components/AsyncButton.test.tsx

```diff
@@ -0,0 +1,13 @@
+import { it } from 'vitest'
+import AsyncButton from 'src/components/AsyncButton'
+import { render, screen } from '@testing-library/react'
+
+it('renders a spinner when loading', async () => {
+  render(<AsyncButton loading={true} />)
+  expect(screen.getByRole('status')).toBeTruthy()
+})
+
+it('disables button when loading', async () => {
+  const { baseElement } = render(<AsyncButton loading={true} disabled />)
+  expect(baseElement.querySelector('button')).toBeTruthy()
+})
```

Comment:
1. We want to check that the button automatically disables itself when loading, so lets remove the hardcoded `disabled` here on line 11.
2. Line 12 should check that the button is disabled with the selector `button:disabled` rather than just `button`.